optuna/samplers/_tpe/sampler.py

```diff
@@ -827,3 +829,40 @@ def _get_multivariate_observation_pairs(
             values[param_name].append(param_value)
 
     return values, scores
+
+
+def _filter_out_observation_pairs(
+    values: List[Optional[float]],
+    scores: List[Tuple[float, float]],
+    distribution: BaseDistribution,
+) -> Tuple[List[Optional[float]], List[Tuple[float, float]]]:
+    ret_values = []
+    ret_scores = []
+    for value, score in zip(values, scores):
+        if value is None or distribution._contains(value):
```

Comment:
I think `value` should not be `None`. Otherwise, all the observations will be `None` and TPE algorithm will not work as expected.
```suggestion
        if value is not None or distribution._contains(value):
```